src/modules/forum/sagas.ts

```diff
@@ -20,17 +21,31 @@ export function* forumSaga(builder: BuilderAPI) {
     const { collection, forumPost } = action.payload
     try {
       const forumLink: string = yield call([builder, 'createCollectionForumPost'], collection, forumPost)
-      return put(createCollectionForumPostSuccess(collection, forumLink))
+      yield put(createCollectionForumPostSuccess(collection, forumLink))
```

Comment:
Although yield works here, return is correct, as it's the last step of an iterator.

Reply:
On the added case, `return put` was not working as it was not being handled. Had to write it as `yield put; return` for it to work. Might be a thing with sagas. I used yield everywhere to have a consistent usage in the saga.